clippy_lints/src/missing_const_for_fn.rs

```diff
@@ -150,7 +152,7 @@ impl<'tcx> LateLintPass<'tcx> for MissingConstForFn {
             if cx.tcx.is_const_fn_raw(def_id.to_def_id()) {
                 cx.tcx.sess.span_err(span, err.as_ref());
             }
-        } else {
+        } else if !is_from_proc_macro_and_span(cx, &kind, span) {
```

Comment:
So as far as I was reading your PR, this should be done as late as possible because it's very inefficient. I hope this is the right approach.

Reply:
It's slower than matching on HIR or AST nodes. This could be done before the MIR check, but it's fine either way.

Reply:
I'm not entirely sure where the MIR check is. Do you mean before `LateLintPass`? How would that work?

Reply:
Before the call to `optimized_mir` right above this.

Reply:
Done.